_data-prepper/trace-analytics.md

```diff
@@ -0,0 +1,380 @@
+
+#### Heap
+
+Configure the heap of Data Prepper by setting the `JVM_OPTS` environmental variable. We recommend that you set the heap value to a minimum value of `4` * `batch_size` * `otel_send_batch_size` * `maximum size of indvidual span`.
+
+As mentioned in the [setup guide]({{site.url}}{{site.baseurl}}/data-prepper/trace_analytics/#opentelemetry-collector), set `otel_send_batch_size` to a value of `50` in your OpenTelemetry collector configuration.
+
+<!--- This link needs to be checked.--->
+
+#### Local disk
+
+Data Prepper uses the local disk to store metadata required for service-map processing, so we recommend only storing the key fields `traceId`, `spanId`, `parentSpanId`, `spanKind`, `spanName` and `serviceName`. The service-map plugin ensures that it only stores two files, with each storing `window_duration` seconds of data. During testing we found that, for a throughput of `3000 spans/second`, the total disk usage was `4 MB`.
+
+Data Prepper also uses the local disk to write logs. In the most recent version of Data Prepper, you can redirect the logs to the path of your preference.
+
+
+### AWS
+
+The [AWS EC2 Cloudformation](https://github.com/opensearch-project/data-prepper/blob/main/deployment-template/ec2/data-prepper-ec2-deployment-cfn.yaml) template provides a user-friendly mechanism to configure the above scaling attributes.
+
+The [Kubernetes configuration files](https://github.com/opensearch-project/data-prepper/blob/main/examples/dev/k8s/README.md) and [EKS configuration files](https://github.com/opensearch-project/data-prepper/blob/main/deployment-template/eks/README.md) are available to configure these attributes in a cluster deployment.
+
+### Benchmark
+
+We ran tests in a `r5.xlarge` EC2 instance with the following configuration:
+ 
+ * `buffer_size`: 4096
+ * `batch_size`: 256
+ * `workers`: 8
+ * `Heap`: 10GB
+ 
+The above setup was able to handle a throughput of `2100` spans/second at `20` percent CPU utilization.
+
+## Pipeline configuration
+
+The following sections describe pipeline configuration.
+
+### Trace analytics pipeline example
+
+The following example demonstrates how to build a pipeline that supports the [OpenSearch Dashboards Observability plugin]({{site.url}}{{site.baseurl}}/observability-plugin/trace/ta-dashboards/). This pipeline takes data from the OpenTelemetry Collector and uses two other pipelines as sinks. These two, separate pipelines index trace and the service map documents for the dashboard plugin.
+
+Starting with Data Prepper version 2.0, Data Prepper no longer supports the `otel_trace_raw_prepper` processor due to the Data Prepper internal data model evolution. Instead, users should use the  `otel_trace_raw` processor. See the following .yaml file example:
```

Comment:
@dlvenable "...due to the Data Prepper internal data model evolution." - Could we change this to "...due to Data Prepper's internal data model structure"? "internal data model evolution" is not very clear.

Reply:
I'm good with the proposed change.